domain/src/main/java/org/oppia/domain/oppialogger/exceptions/OppiaUncaughtExceptionHandler.kt

```diff
@@ -0,0 +1,27 @@
+package org.oppia.domain.oppialogger.exceptions
+
+import org.oppia.util.logging.ConsoleLogger
+import org.oppia.util.system.OppiaClock
+import javax.inject.Inject
+
+/** Handler for catching fatal exceptions before the defaultUncaughtExceptionHandler. */
+class OppiaUncaughtExceptionHandler @Inject constructor(
```

Comment:
We should generally try to avoid classes having generic names. What if we wanted to add more functionality to the uncaught exception handler chain? We could either have an omni OppiaUncaughtExceptionHandler that does lots of things, or a compositional approach where we have multiple exception handlers with specific responsibilities. The latter tends to be more flexible & has other maintenance benefits. Naming this class specific helps move toward that pattern. See the gist in my latest comment for a suggested alternate name.

Reply:
Okay, got it. Name changed to the one suggested.